cl/corpus_importer/management/commands/harvard_opinions.py

```diff
@@ -509,16 +511,21 @@ def add_citations(cites: List[CitationType], cluster_id: int) -> None:
         if not citation[0].canonical_reporter:
             reporter_type = Citation.STATE
         else:
-            cite_type_str = citation[0].exact_editions[0].reporter.cite_type
+            cite_type_str = citation[0].all_editions[0].reporter.cite_type
             reporter_type = map_reporter_db_cite_type(cite_type_str)
 
-        Citation.objects.get_or_create(
-            volume=citation[0].volume,
-            reporter=citation[0].reporter,
-            page=citation[0].page,
-            type=reporter_type,
-            cluster_id=cluster_id,
-        )
+        try:
+            Citation.objects.get_or_create(
+                volume=citation[0].volume,
+                reporter=citation[0].reporter,
+                page=citation[0].page,
+                type=reporter_type,
+                cluster_id=cluster_id,
+            )
+        except IntegrityError:
+            logger.warning(
+                f"Reporter mismatch for cluster: {cluster_id} on cite: {cite['cite']}"
```

Comment:
I'm confused what causes this or what the fix is? More detail, please?

Reply:
This is for situations where we have 900 incorrect citation types in the future - that we can identify and reprocess after the fact - but because I dont know how many incorrectly scraped opinions exist I dont want to crash every time we come across them in future reporter imports.  This way I can log it- find them and fix them and reprocess it